scripts/travis/upload-to-registry.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+set -euxf -o pipefail
+
+usage() {
+  echo $"Usage: $0 <image>"
+  exit 1
+}
+
+check_args() {
+  if [ ! $# -eq 1 ]; then
+    echo "ERROR: need exactly one argument"
+    usage
+  fi
+}
+
+compute_image_tag() {
+  local branch=$1
+
+  if [[ "${branch}" == "master" ]]; then
+    local tag="latest"
+    echo ${tag}
+  elif [[ "${branch}" =~ ^v([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
+    local major="${BASH_REMATCH[1]}"
+    local minor="${BASH_REMATCH[2]}"
+    local patch="${BASH_REMATCH[3]}"
+    local tag=${major}.${minor}.${patch}
+    echo "${tag} ${major} ${minor} ${patch}"
+  else
+    local tag="${branch}"
+    echo ${tag}
+  fi
+}
+
+label_release_tag() {
+  local registry=$1
+  local image=$2
+  local major=${3:-}
+  local minor=${4:-}
+  local patch=${5:-}
+
+  if [[ -n ${major} ]]; then
+    docker tag ${image} ${registry}/${image}:${major}
+    if [[ -n ${minor} ]]; then
+      docker tag ${image} ${registry}/${image}:${major}.${minor}
+      if [[ -n ${patch} ]]; then
+        docker tag ${image} ${registry}/${image}:${major}.${minor}.${patch}
+      fi
+    fi
+  fi
+}
+
+try_login() {
+  local registry=$1
+  local user=$2
+  local token=$3
+  local marker=$4
+
+  if [ ! -f ${marker} ]; then
```

Comment:
```suggestion
  if [ ! -f ${marker} && -v ${token} ]; then
```

(assuming that `&&` syntax is correct). This makes the script tolerant when one of the registry tokens is not passed.